spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/PropertiesLauncher.java

```diff
@@ -489,7 +499,7 @@ private Archive getArchive(File file) throws IOException {
 	}
 
 	private List<Archive> getNestedArchives(String root) throws Exception {
-		if (root.startsWith("/")
+		if (!root.equals("/") && root.startsWith("/")
 				|| this.parent.getUrl().equals(this.home.toURI().toURL())) {
 			// If home dir is same as parent archive, no need to add it twice.
 			return null;
```

Comment:
Thanks, Dave. We agreed on Slack not to merge 0e8f05542f7d so I'd be ignoring those integration tests. The new ones are better, but I don't like them being part of `spring-boot-gradle-tests` because they're not really anything to do with Gradle. Also, that module doesn't exist any more in 2.0 so they'll be a pain to merge in their current form. Is it not possible to integration tests this in the spring-boot-loader module by creating a jar or two with the required layout in code?